src/Compilers/Core/Portable/Syntax/SyntaxRemoveOptions.cs

```diff
@@ -46,6 +46,12 @@ public enum SyntaxRemoveOptions
         /// <summary>
         /// Adds elastic marker trivia
         /// </summary>
-        AddElasticMarker = 0x20
+        AddElasticMarker = 0x20,
+
+        /// <summary>
+        /// Use the same end of line used in the node being removed, when adding one.
+        /// If no end of line was present, use default.
+        /// </summary>
+        UseOriginalEndOfLine = 0x40
```

Comment:
can we set default?

Reply:
With the current state of this PR we can't. Currently, there is only [one scenario](https://github.com/dotnet/roslyn/pull/23165/commits/dc5568b3f3a3dfb4a67ced53f410d29c8dd5ae69#diff-0e0e5cee589dd5835bc357a2c0406c46R2511) that will force the node remover to use a default EOL. If we would change the current [IsEndOfLine](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs#L120) method to return `false` when a single line comment trivia is the last line in the document, the EOL would always be found and we wouldn't need to use a default EOL.

Reply:
One possible solution in order to avoid using a default EOL is to change the `IsEndOfLine` to something like:
```
private static bool IsEndOfLine(SyntaxTrivia trivia)
{
  return trivia.Kind() == SyntaxKind.EndOfLineTrivia
    // A single line comment represents an EOL only if it is not the last trivia in the document
    || (trivia.Kind() == SyntaxKind.SingleLineCommentTrivia && trivia.Span.End != trivia.SyntaxTree.Length)
    || trivia.IsDirective;
}
```
but this would affect also the `KeepEndOfLine` option, when a single line comment is the last trivia in the document. If you are ok with that, I can include it in this PR.